### PR TITLE
[Feature] Add support of points_in_polyogns for Ascend device

### PR DESCRIPTION
**Pull request by long11111111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: tests/test_ops/test_points_in_polygons.py [hunk 1]
```diff
@@ -21,3 +22,35 @@ def test_points_in_polygons():
...
+    pytest.param(
+        'mlu',
+        marks=pytest.mark.skipif(
+            not IS_MLU_AVAILABLE, reason='requires MLU support')),
+    pytest.param(
+        'npu',
+        marks=pytest.mark.skipif(
+            not IS_NPU_AVAILABLE, reason='requires NPU support'))
+])
+def test_points_in_polygons_device(device):
```
Comment (zhouzaida) [inline tests/test_ops/test_points_in_polygons.py:41]:
```suggestion
def test_points_in_polygons(device):
```

Comment (zhouzaida) [reply tests/test_ops/test_points_in_polygons.py:41]: This test (https://github.com/open-mmlab/mmcv/pull/2848/files#diff-efebcc8384b4e339d2672c04d405af66df075e5e231e29bab2571442cb0545d4R12) can be removed.

File: tests/test_ops/test_points_in_polygons.py [hunk 2]
```diff
@@ -21,3 +22,35 @@ def test_points_in_polygons():
...
+
+@pytest.mark.parametrize('device', [
+    pytest.param(
+        'cuda',
+        marks=pytest.mark.skipif(
+            not IS_CUDA_AVAILABLE, reason='requires CUDA support')),
+    pytest.param(
+        'mlu',
+        marks=pytest.mark.skipif(
+            not IS_MLU_AVAILABLE, reason='requires MLU support')),
```
Comment (zhouzaida) [inline tests/test_ops/test_points_in_polygons.py:35]: MLU does not support this op so these lines can be removed.

File: mmcv/ops/points_in_polygons.py [hunk 3]
```diff
@@ -31,8 +31,14 @@ def points_in_polygons(points: Tensor, polygons: Tensor) -> Tensor:
...
+    if points.device.type == 'npu':
+        output = torch.full([points.shape[0], polygons.shape[0]],
+                            0.).to(torch.float32).npu()
+        ext_module.points_in_polygons_forward(points.contiguous(),
+                                              polygons.contiguous(), output)
+    else:
+        output = torch.full([points.shape[0], polygons.shape[0]],
+                            0.).cuda().float()
+        ext_module.points_in_polygons_forward(points.contiguous(),
+                                              polygons.contiguous(), output)
```
Comment (zhouzaida) [inline mmcv/ops/points_in_polygons.py:43]:
```suggestion
    output = torch.zeros(points.shape[0], polygons.shape[0],
                         dtype=torch.float32, device=points.device)
    ext_module.points_in_polygons_forward(points.contiguous(),
                                          polygons.contiguous(), output)
```

File: tests/test_ops/test_points_in_polygons.py (hunk 1 above)
Comment (long11111111) [reply tests/test_ops/test_points_in_polygons.py]: modified according to suggestions

File: tests/test_ops/test_points_in_polygons.py (hunk 2 above)
Comment (long11111111) [reply tests/test_ops/test_points_in_polygons.py]: modified according to suggestions

File: mmcv/ops/points_in_polygons.py (hunk 3 above)
Comment (long11111111) [reply mmcv/ops/points_in_polygons.py]: modified according to suggestions

**Review by zhouzaida — APPROVED** [review]

**Comment by zhouzaida** [comment]
Hi @long11111111 , lint failed.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**